lib/hotwire_combobox/helper.rb

```diff
@@ -115,7 +115,7 @@ def hw_first_page?
 
       def hw_uri_with_params(url_or_path, **params)
         URI.parse(url_or_path).tap do |url_or_path|
-          query = URI.decode_www_form(url_or_path.query || "").to_h.merge(params.compact_blank.stringify_keys)
+          query = Rack::Utils.parse_query(url_or_path.query || "").merge(params.compact_blank.stringify_keys)
```

Comment:
I didn't know Rack exposed this util. Nice!

Glancing at Rack's source code, it seems we should use `parse_nested_query` instead. Any reason to prefer `parse_query` over that?

Also, I noticed Rack exposes `build_nested_query` as well. What do you think about using that instead of `URI.encode_www_form` on the next line? Probably a good idea to use the pair together as they'll likely evolve together as well inside Rack.